tests/test_conditional_steps.py

```diff
@@ -1,40 +1,89 @@
 import re
+import os
 import inspect
-import pytest
 
 from universum import __main__
+from universum.configuration_support import Step
 
 
-true_branch_step_name = "true_branch"
-false_branch_step_name = "false_branch"
+class StepsInfo:
+    conditional_step = None
+    true_branch_step = None
+    false_branch_step = None
+    is_conditional_step_passed = False
 
 
 def test_conditional_true_branch(tmpdir, capsys):
-    check_conditional_step_success(tmpdir, capsys, conditional_step_passed=True)
+    steps_info = get_conditional_steps_info(is_conditional_step_passed=True)
+    check_conditional_step(tmpdir, capsys, steps_info)
 
 
 def test_conditional_false_branch(tmpdir, capsys):
-    check_conditional_step_success(tmpdir, capsys, conditional_step_passed=False)
+    steps_info = get_conditional_steps_info(is_conditional_step_passed=False)
+    check_conditional_step(tmpdir, capsys, steps_info)
 
 
-def check_conditional_step_success(tmpdir, capsys, conditional_step_passed):
-    config_file = build_config_file(tmpdir, conditional_step_passed)
-    check_conditional_step(tmpdir, capsys, config_file, conditional_step_passed)
+# https://github.com/Samsung/Universum/issues/744
+# Artifact will be collected only from the second executed step, the first one will be overwritten
+# Skipping checking file content to not overload tests with additional logic for incorrect behaviour check
+def test_same_artifact(tmpdir, capsys):
+    steps_info = get_conditional_steps_info(is_conditional_step_passed=True)
 
+    conditional_step_artifact = steps_info.conditional_step["artifacts"]
+    steps_info.true_branch_step.command = ["touch", conditional_step_artifact]
+    steps_info.true_branch_step.artifacts = conditional_step_artifact
 
-def build_config_file(tmpdir, conditional_step_passed):
-    conditional_step_exit_code = 0 if conditional_step_passed else 1
+    check_conditional_step(tmpdir, capsys, steps_info)
 
+
+def test_no_artifact_in_executed_branch_step(tmpdir, capsys):
```

Comment:
Is the logic of this test to check that artifacts are not collected if they are not specified for step?
How does it relate to conditional steps? 
If it doesn't, I think this is a test for basic functionality of artifacts collection and should be a part of its tests.
If the idea is to test that artifacts are not collected for the opposing branch (fail for successful case and success for failure case), then the test is incorrect.